UIKitTests/CALayerTests.swift

```diff
@@ -69,4 +69,88 @@ class CALayerTests: XCTestCase {
         XCTAssertEqual(layer.frame.origin.x, expectedSize.width, accuracy: accuracy)
         XCTAssertEqual(layer.frame.origin.y, expectedSize.height, accuracy: accuracy)
     }
+
+    func testLayoutSuperlayerDelegateWhenChangingBoundsSize() {
+        class TestView: UIView {
+            var onLayoutSubviews: (() -> ())?
+            override func layoutSubviews() {
+                super.layoutSubviews()
+                onLayoutSubviews?()
+            }
+        }
+
+        let superView = TestView()
+        let view = TestView()
+        superView.addSubview(view)
+        view.layer.bounds.size = CGSize(width: 10, height: 10)
+
+        superView.layoutIfNeeded()
+        view.layoutIfNeeded()
+
+        view.layer.bounds.size = CGSize(width: 15, height: 15)
```

Comment:
interestingly it does also relayout the superlayer when mutating `view.layer.bounds.origin`, but I think we should not mimic this behaviour because this also means that we relayout superlayers of UIScrollViews on every scrolled pixel which is not the case on iOS! (see the test below).

There seems to be a different connection between `contentOffset` and `bounds.origin` between ios and our UIKit 

@ephemer

Reply:
I think we'd need to discuss this point on a call – not entirely sure what you mean about the scroll view right now. Is it possible to prevent the infinite layouts on the scrollview by changing that layer's delegate somehow?

I thought we only do a re-layout when the `bounds.size` changes though, so I don't understand how `contentOffset` or `bounds.origin` are related?

Reply:
Setting contentOffset directly changes the bounds.origin of the scrollviews layer. On iOS and in our uikit - but I agree, could be more efficient to discuss this in a call. I'll send out an invite for next week.

- Anyways, the pr is ready for being merged. But I would like to share my findings about scrolllview bounds with you

Reply:
@michaelknoch the behaviour we're talking about is only related to `bounds.size`, no? How is that related to bounds.origin / contentOffset?